packages/app/src/utils/rsc.ts

```diff
@@ -1,3 +1,5 @@
 export const getSegment = (props: any) => {
-  return props.children.props.childProp.segment;
+  console.log('SEGMENT=', props.children?.props?.childProp?.segment);
+  // FIX (JL,2024-07-05): This deref chain always results in `undefined`. How to fix?
+  return props.children?.props?.childProp?.segment;
```

Comment:
I don't understand this problem, other than it seems to have something to do with Routes and Segments. This remains an issue,